Small changes to the README

j-kleemann opened the pull request:
Fixed a few typos and tried to clarify a few things

uga-uga commented:
Good idea to clarify the synonymous use of the words hit/step. Nevertheless, some exchanges of the words change the logics.
For example, the sentence 
```
Note that a "step" in the GEANT4 sense does not necessarily imply an interaction with the sensitive detector ...
```
is more misleading now than before. Indeed, a 'step' does not imply an interaction. However, even a 'hit' does not imply an interaction, because hits may also denote volume crossings. This alludes to the difference between G4Hit and G4Step

j-kleemann commented:
I exchanged the occurrences of "step" for "hit" in the relevant cases.
I wasn't aware of the difference between "steps" and "hits" before, sorry for that.
To my understanding "hits" are produced "when something happens", while the "steps" simply connect the "hits".
As steps never seem to be of relevance for an average utr user, I now removed them from the README to again achieve uniformity of it.
I hope this addresses your concerns
